lib/Runtime/PlatformAgnostic/Platform/Linux/UnicodeText.ICU.cpp

```diff
@@ -283,25 +274,41 @@ namespace PlatformAgnostic
             return true;
         }
 
-#define EMPTY_COPY \
-    const int len = (destLength <= sourceLength) ? destLength - 1 : sourceLength; \
-    memcpy(destString, sourceString, len * sizeof(char16)); \
-    destString[len] = char16(0); \
-    *pErrorOut = NoError; \
-    return len;
+#define minm(a,b) ((a < b) ? a : b)
 
         int32 NormalizeString(NormalizationForm normalizationForm, const char16* sourceString, uint32 sourceLength, char16* destString, int32 destLength, ApiError* pErrorOut)
         {
-            // TODO: implement this
-            EMPTY_COPY
+            *pErrorOut = ApiError::NoError;
+            if (destString == nullptr)
+            {
+                return sourceLength;
+            }
+
+            int32 len = (int32) minm(minm(destLength, sourceLength), INT_MAX);
+            memcpy(destString, sourceString, len * sizeof(char16));
+
+            return len;
         }
 
+typedef WCHAR (*ToUPorLOW)(WCHAR);
```

Comment:
nit: This name confused me for a while because I was expecting CamelCase. maybe ToUpOrLow, or maybe just CaseConversionFunc?

Reply:
well, it's a local typedef. Can we consider as it's just distinct?